File: C/ch3-多元3-遞減氣泡排序.c
#include <stdio.h>

int main(){
    int a[6] = {3, 8, 4, 2, 5, 7};
    int i, j, tmp, k;

    for(i=1; i<6; i++){
        for(j=0; j<6-i; j++){
            if(a[j] < a[j+1]){
                tmp = a[j];
                a[j] = a[j+1];
                a[j+1] = tmp;
            }
        }
        printf("第%d回合結果:", i);
        for (k=0;k<6;k++)
            printf("%d ", a[k]);
        printf("\n");
    }

    return 0; 
}

File: Python/ch3_多元3_遞減氣泡排序.py
a = [3, 8, 4, 2, 5, 7]
for i in range(1, 6):
  for j in range(6-i):
    if a[j] < a[j+1]:
      tmp = a[j]
      a[j] = a[j+1]
      a[j+1] = tmp
  print('第',i,'回合結果:', end='')
  for k in range(6):
    print(a[k], end='')
  print()